trie/utils/verkle.go

```diff
@@ -289,12 +267,17 @@ func GetTreeKeyStorageSlotTreeIndexes(storageKey []byte) (*uint256.Int, byte) {
 	}
 	// If the storage slot is in the main storage, we need to add the main storage offset.
 
+	// Get the new offset since we now know that we are above 64.
+	pos.Sub(&pos, codeStorageDelta)
+	suffix := byte(pos[0] & 0xFF)
+
```

Comment:
subtract the code storage from the offset before adding the main storage offset value. Get the last byte, because what follows is a right shift.